drivers/flash/flash_stm32_ospi.c

```diff
@@ -351,10 +352,13 @@ static int stm32_ospi_read_jedec_id(const struct device *dev)
 	OSPI_RegularCmdTypeDef cmd = ospi_prepare_cmd(OSPI_SPI_MODE, OSPI_STR_TRANSFER);
 
 	cmd.Instruction = JESD216_CMD_READ_ID;
-	cmd.DummyCycles = 8U;
 	cmd.AddressSize = HAL_OSPI_ADDRESS_NONE;
 	cmd.NbData = JESD216_READ_ID_LEN; /* 3 bytes in the READ ID */
 
+	if (dev_cfg->data_mode == OSPI_OPI_MODE) {
+		cmd.DummyCycles = 8U;
```

Comment:
According to the macronix [MX25UM51245] octoflash, the dummy cycles are required for reading RDID in DTR-OPI Mode. 
--> STR mode (OPI orSPi does not require dummy cycles)

Reply:
If cmd is prepared for SPI/STR that doesn't need any dummy cycles and failed with 8. But 8 are required for OPI/DTR. That would mean it never worked for both modes.

Considering #47962, how did it work for mentioned [mx25lm51245](https://www.macronix.com/Lists/Datasheet/Attachments/8729/MX25LM51245G,%203V,%20512Mb,%20v1.1.pdf) (_seem to require 4 dummy cycles for STR-OPI mode_) or [macronix MX25L51245G datasheet](https://media.digikey.com/pdf/Data%20Sheets/Macronix/MX25L51245G.pdf) (_supports only SPI/STR with 0 dummy_)?